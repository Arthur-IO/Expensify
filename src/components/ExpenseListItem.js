import React from "react";

import {Link} from "react-router-dom";

const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
        <h3>
            {/* Links this specific component to the link provided */}
            <Link to={`/edit/${id}`}>
                {description}
            </Link>
        </h3>
        <p>{amount} - {createdAt}</p>
    </div>
)

export default ExpenseListItem