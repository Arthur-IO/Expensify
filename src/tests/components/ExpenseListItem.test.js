import React from "react";
import {shallow} from 'enzyme'
import ExpenseListItem from "../../components/ExpenseListItem";

test('should render ExpenseListItem with provided information', () => {
    const wrapper = shallow(
        <ExpenseListItem
            id={2352345}
            description={"Orange soda"}
            amount={300}
            createdAt={12341324423}
        />
    )
    expect(wrapper).toMatchSnapshot()
})