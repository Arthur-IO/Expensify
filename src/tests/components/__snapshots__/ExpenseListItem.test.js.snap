// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with provided information 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/2352345"
    >
      Orange soda
    </Link>
  </h3>
  <p>
    $3.00
    -
    May 23rd 1970
  </p>
</div>
`;
